fix: adding lists to core

mathisscott opened the pull request:
• added global css for ol and ul elements
• created SASS module for lists
• fixed minor issue with vertical layout attr selector
• encapsulated all list styles within `cds-list` attr
• created styles for ordered, unordered, and unsettled lists
• created styles to handle nested lists
• created ordered attr values for ~50 regions listed as supported on MDN
• made them different values b/c we can't update a CSS var on a non-root scope in IE #ThanksAgainIE
• users may also want more granular typings that would not be feasible with global CSS vars
• once we drop IE support (sometime in 2100 AD?) we can refactor these styles to use a single CSS var
• the list of regions will still be useful tho

Signed-off-by: Scott Mathis <smathis@vmware.com>

Commit:
fix: adding lists to core

• added global css for ol and ul elements
• created SASS module for lists
• fixed minor issue with vertical layout attr selector
• encapsulated all list styles within `cds-list` attr
• created styles for ordered, unordered, and unsettled lists
• created styles to handle nested lists
• created ordered attr values for ~50 regions listed as supported on MDN
• made them different values b/c we can't update a CSS var on a non-root scope in IE #ThanksAgainIE
• users may also want more granular typings that would not be feasible with global CSS vars
• once we drop IE support (sometime in 2100 AD?) we can refactor these styles to use a single CSS var
• the list of regions will still be useful tho

Signed-off-by: Scott Mathis <smathis@vmware.com>

coryrylan commented on src/clr-core/styles/list/_list.scss:
```diff
@@ -0,0 +1,73 @@
+
+@include renderListStyles(ol, arabic-indic armenian bengali cjk-earthly-branch cjk-heavenly-stem cjk-ideographic decimal decimal-leading-zero devanagari ethiopic-halehame-am ethiopic-halehame-ti-er ethiopic-halehame-ti-et georgian gujarati gurmukhi hangul hangul-consonant hiragana hiragana-iroha kannada katakana katakana-iroha khmer korean-hangul-formal korean-hanja-formal korean-hanja-informal l ...
+
+ul[cds-list='unstyled'] {
```
If the majority of ordered list styles is based on regions couldn't this be a root variable that the app sets based on the region they are serving? If its on root it should work with the IE polyfill and then we don't have to send region specific styles to every client.

```scss
// _list.scss
ol[cds-list] {
  list-style-type: var(--cds-ordered-list-style-type, decimal);
}
```

```scss
// app css dev can override 
:root {
  --cds-ordered-list-style-type: thai;
}
```

Or if that doesnt work seems like region specific styles could be overridden globally for the entire application similar to how we handle i18n globally.

```scss
// app dev adds this css based on their supported region
ol[cds-list] {
  list-style-type: thai;
}
```

Commit:
fix: pr updates

• added tokens for default list types
• added fix for list items inside of a horizontal layout
• removed regional values from list attr
• added demos of lists using vertical and horizontal layouts
• added demo of a regional list type using custom CSS

Signed-off-by: Scott Mathis <smathis@vmware.com>

Commit:
fix: pr fixup

Signed-off-by: Cory Rylan <crylan@vmware.com>
